File: backend/internal/model/email/application/smtp_settings.go
package emailapplication

// PublicSMTPConfiguration is what the admin-facing API and its callers are
// allowed to know about a stored configuration: everything except the
// password, represented only by PasswordConfigured.
type PublicSMTPConfiguration struct {
	Host               string
	Port               uint16
	TLSMode            TLSMode
	Authentication     AuthenticationMode
	Username           string
	FromAddress        string
	PasswordConfigured bool
}

// SMTPSettings is the settings-read result. A nil Configuration is the valid
// unconfigured state.
type SMTPSettings struct {
	Configuration *PublicSMTPConfiguration
}

// Public projects cfg into its secret-free representation. A nil cfg (no
// configuration stored) projects to a nil *PublicSMTPConfiguration.
func Public(cfg *SMTPConfiguration) *PublicSMTPConfiguration {
	if cfg == nil {
		return nil
	}
	return &PublicSMTPConfiguration{
		Host:               cfg.Host,
		Port:               cfg.Port,
		TLSMode:            cfg.TLSMode,
		Authentication:     cfg.Authentication,
		Username:           cfg.Username,
		FromAddress:        cfg.FromAddress,
		PasswordConfigured: cfg.Password != "",
	}
}

File: frontend/src/model/email/api/smtpSettings.ts
// Wire-format boundary types for the admin SMTP settings API. This mirrors
// the backend's model/email/api package - the field names and enum wire
// values are decided there; this file only reflects that process boundary.
import type { AuthenticationMode, TLSMode } from "../../../config/constants/email-providers.ts";

export interface PublicSMTPConfiguration {
  host: string;
  port: number;
  tlsMode: TLSMode;
  authentication: AuthenticationMode;
  username: string;
  fromAddress: string;
  passwordConfigured: boolean;
}

export interface SMTPSettings {
  configured: boolean;
  configuration?: PublicSMTPConfiguration;
}

export interface SaveSMTPSettingsRequest {
  host: string;
  port: number;
  tlsMode: TLSMode;
  authentication: AuthenticationMode;
  username: string;
  password?: string;
  fromAddress: string;
}
